lib/ansible/module_utils/known_hosts.py

```diff
@@ -115,15 +115,13 @@ def not_in_host_file(self, host):
             continue
 
         try:
-            host_fh = open(hf)
+            with open(hf) as host_fh:
+                data = host_fh.readlines()
```

Comment:
Same here. You are changing behavior.
```suggestion
                data = host_fh.read()
```